### PR TITLE
Adding functionality for within province analysis, and an example function for plotting over time

### PR DESCRIPTION
**Pull request by kylemath**
I think I took care of all changes needed to add a province search to the database functions, as well as an example of how to plot provincial / state changes over time in canada, I updated the readme as well

as a bonus I also changed the colours scale of the map to be more intuitive

as a bonus bonus I also fixed the bug with China's name in https://github.com/sudharshan-ashok/pycovid/issues/2

### CODE REVIEW

**Comment by kylemath** [comment]
This can make plots like this: 
Screen Shot 2020-03-14 at 3 28 44 PM

File: pycovid/pycovid.py [hunk 1]
```diff
@@ -12,12 +12,23 @@ def getCovidCases(countries=None, start_date=None, end_date=None, casetype=['con
...
+            if province not in df.province_state.values:
+                print("Province: {0} not found in database. Check spelling!".format(province))
+            df = df[(df.province_state.isin(provinces))] 
             
     if countries is not None:
         for country in countries:
             if country not in df.country_region.values:
                 print("Country: {0} not found in database. Check spelling!".format(country))
             df =  df[(df.country_region.isin(countries))]
+
```
Comment (kylemath) [inline pycovid/pycovid.py:27]: oh I also added this cumsum option to the getCovidCases function

File: pycovid/pycovid.py [hunk 2]
```diff
@@ -121,7 +132,30 @@ def plot_countries(df=None, grouped_data=False, metric="confirmed"):
...
                     color=metric, # lifeExp is a column of gapminder
                     hover_name="country_region", # column to add to hover information
-                    color_continuous_scale='RdBu_r')
+                    color_continuous_scale="OrRd")
     fig.show()
     
+
+def plot_provinces(countries=None, provinces=None, start_date=None, end_date=None, casetype=['confirmed', 'death', 'recovered']):
+    
+    # populations =   Ontario   14,446,515
```
Comment (kylemath) [inline pycovid/pycovid.py]: Leaving this here so I can add a feature to estimate proportion of population,

**Review by sudharshan-ashok — APPROVED** [review]
The changes look OK and the new functions are great!

**Comment by sudharshan-ashok** [comment]
Check the new version 0.1.1 on pyPI

File: pycovid/pycovid.py [hunk 3]
```diff
@@ -112,7 +123,7 @@ def getIntervalData(df, interval='30D'):
 
 def plot_countries(df=None, grouped_data=False, metric="confirmed"):
     
-    if df is None:
+    if df == None:
```
Comment (GaelVaroquaux) [inline pycovid/pycovid.py:126]: Actually, testing for "None" is better via "is". "None" is a singleton in Python, while "==" can do all kind of surprising things (eg on numpy arrays)

File: pycovid/pycovid.py [hunk 4]
```diff
@@ -81,7 +92,7 @@ def getCovidCasesWide(countries=None, start_date=None, end_date=None, casetype=[
     iso_df = pd.read_csv('https://raw.githubusercontent.com/lukes/ISO-3166-Countries-with-Regional-Codes/master/slim-3/slim-3.csv')
     iso_df = iso_df[['name', 'alpha-3']]
     iso_df.loc[iso_df.name=="United States of America", 'name'] = 'US'
-    iso_df.loc[iso_df.name=="China", 'name'] = 'Mainland China'
+    iso_df.loc[iso_df.name=="China", 'name'] = 'China'
```
Comment (GaelVaroquaux) [inline pycovid/pycovid.py:95]: Hum, it seems that this didn't solve the problem with getting the Chinese data. Did it?

Comment (GaelVaroquaux) [reply pycovid/pycovid.py:95]: OK, it solved it for getCovidCasesWide, but not getCovidCases

Comment (kylemath) [reply pycovid/pycovid.py:95]: https://github.com/sudharshan-ashok/pycovid/pull/7

File: pycovid/pycovid.py (hunk 3 above)
Comment (kylemath) [reply pycovid/pycovid.py:126]: https://github.com/sudharshan-ashok/pycovid/pull/7

**Pull request merged**